packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -173,7 +173,7 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
     name: 'vite:build-import-analysis',
     resolveId(id) {
       if (id === preloadHelperId) {
-        return id
+        return { id, moduleSideEffects: false }
```

Comment:
Is this change required? Just adding `/* @__PURE__ */` seems to work.

Reply:
The two changes either can work. But I think a little bit redundancy design could be better.

Reply:
OK. Would you move `moduleSideEffects: false` to the `load` hook? I think that make it more easier to understand since whether a file has a side-effect or not depends on the content.